src/components/Navbar/SearchDrawer/SearchDrawer.tsx

```diff
@@ -53,32 +64,68 @@ const SearchDrawer: React.FC = () => {
-  useEffect(() => {
-    // only when the search query has a value we call the API.
-    if (debouncedSearchQuery) {
-      dispatch({ type: addSearchHistoryRecord.type, payload: debouncedSearchQuery });
-      logTextSearchQuery(debouncedSearchQuery, SearchQuerySource.SearchDrawer);
+  /**
+   * Call BE to fetch the results using the passed filters.
+   *
+   * @param {string} query
+   * @param {number} page
+   * @param {string} language
+   */
+  const getResults = useCallback(
+    (query: string, page: number) => {
       setIsSearching(true);
+      logTextSearchQuery(query, SearchQuerySource.SearchDrawer);
       getSearchResults({
-        query: debouncedSearchQuery,
-        ...(selectedTranslations &&
-          !!selectedTranslations.length && {
-            filterTranslations: selectedTranslations.join(','),
-          }),
+        query,
+        size: PAGE_SIZE,
+        page,
+        ...(!!selectedTranslations?.length && {
+          filterTranslations: selectedTranslations.join(','),
+        }), // translations will be included only when there is a selected translation
       })
-        .then((response) => {
+        .then(async (response) => {
           if (response.status === 500) {
             setHasError(true);
           } else {
-            setSearchResult(response);
+            setSearchResult({ ...response, service: SearchService.QDC });
+            const noQdcResults =
+              response.pagination.totalRecords === 0 && !response.result.navigation.length;
             // if there is no navigations nor verses in the response
-            if (response.pagination.totalRecords === 0 && !response.result.navigation.length) {
+            if (noQdcResults) {
               logEmptySearchResults({
-                query: debouncedSearchQuery,
+                query,
                 source: SearchQuerySource.SearchDrawer,
+                service: SearchService.QDC,
+              });
+              const kalimatResponse = await getNewSearchResults({
+                mode: SearchMode.Advanced,
+                query,
+                size: PAGE_SIZE,
+                page,
+                exactMatchesOnly: 0,
+                // translations will be included only when there is a selected translation
+                ...(!!selectedTranslations?.length && {
+                  filterTranslations: selectedTranslations.join(','),
+                  translationFields: 'resource_name',
+                }), // translations will be included only when there is a selected translation
               });
+              setSearchResult({
+                ...kalimatResponse,
+                service: SearchService.KALIMAT,
+              });
+              if (kalimatResponse.pagination.totalRecords === 0) {
+                logEmptySearchResults({
+                  query,
+                  source: SearchQuerySource.SearchDrawer,
+                  service: SearchService.KALIMAT,
+                });
```

Comment:
This is an almost identical code to the one in the search page. We should extract it to a common place used by both to:

1. Keep it DRY
2. Easier future refactoring